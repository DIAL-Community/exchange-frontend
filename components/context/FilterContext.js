import { createContext, useEffect, useState } from 'react'
import { useRouter } from 'next/router'

const FilterContext = createContext()
const FilterDispatchContext = createContext()

const FilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')
  // Task tracker context only
  const [showFailedOnly, setShowFailedOnly] = useState(false)
  const [showGovStackOnly, setShowGovStackOnly] = useState(false)
  const [showMature, setShowMature] = useState(false)

  const [categoryTypes, setCategoryTypes] = useState([])
  const [sdgs, setSdgs] = useState([])
  const [useCases, setUseCases] = useState([])
  const [workflows, setWorkflows] = useState([])

  const [origins, setOrigins] = useState([])
  const [countries, setCountries] = useState([])
  const [sectors, setSectors] = useState([])
  const [organizations, setOrganizations] = useState([])
  const [tags, setTags] = useState([])
  const [datasetTypes, setDatasetTypes] = useState([])

  const [aggregators, setAggregators] = useState([])
  const [operators, setOperators] = useState([])
  const [services, setServices] = useState([])

  const [years, setYears] = useState([])

  const [products, setProducts] = useState([])

  const [buildingBlocks, setBuildingBlocks] = useState([])
  const [showClosed, setShowClosed] = useState(false)

  const [aggregator, setAggregator] = useState(false)
  const [endorser, setEndorser] = useState(false)
  const [endorserLevel, setEndorserLevel] = useState('')

  const [specialties, setSpecialties] = useState([])
  const [certifications, setCertifications] = useState([])

  const [isEndorsed, setIsEndorsed] = useState(false)
  const [endorsers, setEndorsers] = useState([])
  const [licenseTypes, setLicenseTypes] = useState([])
  const [isLinkedWithDpi, setIsLinkedWithDpi] = useState(false)
  const [showDpgaOnly, setShowDpgaOnly] = useState(false)

  const [softwareCategories, setSoftwareCategories] = useState([])
  const [softwareFeatures, setSoftwareFeatures] = useState([])
  const [comparedProducts, setComparedProducts] = useState([])
  const [productStage, setProductStage] = useState(null)

  const [showBeta, setShowBeta] = useState(false)

  const router = useRouter()
  useEffect(() => {
    const handleRouteChange = (url) => {
      // Transitioning to other pages (pathname changing).
      if (url.indexOf(router.pathname) < 0) {
        setSearch('')
        setShowFailedOnly(false)
        setShowGovStackOnly(false)
        setShowMature(false)

        setCategoryTypes([])
        setSdgs([])
        setUseCases([])
        setWorkflows([])

        setOrigins([])
        setCountries([])
        setSectors([])
        setOrganizations([])
        setTags([])
        setDatasetTypes([])

        setAggregators([])
        setOperators([])
        setServices([])

        setYears([])

        setProducts([])

        setBuildingBlocks([])
        setShowClosed(false)

        setAggregator(false)
        setEndorser(false)
        setEndorserLevel('')

        setSpecialties([])
        setCertifications([])

        setIsEndorsed(false)
        setEndorsers([])
        setLicenseTypes([])
        setIsLinkedWithDpi(false)
        setShowDpgaOnly(false)

        setSoftwareCategories([])
        setSoftwareFeatures([])
        setComparedProducts([])
        setProductStage('')

        setShowBeta(false)
      }
    }

    router.events.on('routeChangeComplete', handleRouteChange)

    return () => {
      router.events.off('routeChangeComplete', handleRouteChange)
    }
  }, [router.events, router.asPath, router.pathname])

  const valueProps = {
    search,

    showFailedOnly,
    showGovStackOnly,
    showMature,
    categoryTypes,
    sdgs,
    useCases,
    workflows,

    countries,
    datasetTypes,
    organizations,
    origins,
    sectors,
    tags,

    aggregators,
    operators,
    services,

    years,

    products,

    buildingBlocks,
    showClosed,

    aggregator,
    endorser,
    endorserLevel,
    specialties,
    certifications,

    isEndorsed,
    endorsers,
    licenseTypes,
    isLinkedWithDpi,
    showDpgaOnly,
    softwareCategories,
    softwareFeatures,
    comparedProducts,
    productStage,

    showBeta
  }

  const dispatchProps = {
    setSearch,
    setShowFailedOnly,
    setShowGovStackOnly,
    setShowMature,
    setCategoryTypes,
    setSdgs,
    setUseCases,
    setWorkflows,

    setCountries,
    setDatasetTypes,
    setOrganizations,
    setOrigins,
    setSectors,
    setTags,

    setAggregators,
    setOperators,
    setServices,

    setYears,

    setProducts,

    setBuildingBlocks,
    setShowClosed,

    setAggregator,
    setEndorser,
    setEndorserLevel,
    setSpecialties,
    setCertifications,

    setIsEndorsed,
    setEndorsers,
    setLicenseTypes,
    setIsLinkedWithDpi,
    setShowDpgaOnly,
    setSoftwareCategories,
    setSoftwareFeatures,
    setComparedProducts,
    setProductStage,

    setShowBeta
  }

  return (
    <FilterContext.Provider value={{ ...valueProps }}>
      <FilterDispatchContext.Provider value={{ ...dispatchProps }}>
        {children}
      </FilterDispatchContext.Provider>
    </FilterContext.Provider>
  )
}

export { FilterProvider, FilterContext, FilterDispatchContext }
