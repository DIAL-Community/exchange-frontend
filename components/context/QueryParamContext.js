import { createContext, useState } from 'react'

const QueryParamContext = createContext()
const QueryParamContextProvider = ({ children }) => {
  const [interactionDetected, setInteractionDetected] = useState(false)
  return (
    <QueryParamContext.Provider value={{ ...{ interactionDetected, setInteractionDetected } }}>
      {children}
    </QueryParamContext.Provider>
  )
}

export { QueryParamContextProvider, QueryParamContext }
