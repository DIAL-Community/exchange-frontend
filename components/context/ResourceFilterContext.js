import React, { createContext, useState } from 'react'

const ResourceFilterContext = createContext()
const ResourceFilterDispatchContext = createContext()

const ResourceFilterProvider = ({ children }) => {
  const [search, setSearch] = useState('')
  const [resourceTypes, setResourceTypes] = useState([])
  const [resourceTopics, setResourceTopics] = useState([])
  const [resourceCountries, setResourceCountries] = useState([])

  const resourceFilterValues = {
    search,
    resourceTypes,
    resourceTopics,
    resourceCountries
  }
  const resourceFilterDispatchValues = {
    setSearch,
    setResourceTypes,
    setResourceTopics,
    setResourceCountries
  }

  return (
    <ResourceFilterContext.Provider value={resourceFilterValues}>
      <ResourceFilterDispatchContext.Provider value={resourceFilterDispatchValues}>
        {children}
      </ResourceFilterDispatchContext.Provider>
    </ResourceFilterContext.Provider>
  )
}

export {
  ResourceFilterProvider,
  ResourceFilterContext,
  ResourceFilterDispatchContext
}
