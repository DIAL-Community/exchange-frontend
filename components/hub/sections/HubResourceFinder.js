import { FormattedMessage } from 'react-intl'
import HubResources from './HubResources'

const HubResourceFinder = () => {
  return (
    <>
      <img className='h-80 w-full object-cover' alt='DIAL DPI Resource Hub' src='/images/hero-image/hub-hero.svg'/>
      <div className='absolute w-full left-1/2 -translate-x-1/2 min-h-[20rem]' style={{ top: 'var(--ui-header-height)' }}>
        <div className='text-2xl text-center text-dial-cotton pt-20 pb-10 uppercase mx-auto max-w-prose'>
          <FormattedMessage
            id='dpi.resourceFinder.title'
            values={{
              break: () => <br />
            }}
          />
        </div>
        <div className='text-lg text-center text-dial-cotton mx-auto max-w-prose'>
          <FormattedMessage
            id='dpi.resourceFinder.subtitle'
            values={{
              break: () => <br />
            }}
          />
        </div>
      </div>
      <HubResources />
    </>
  )
}

export default HubResourceFinder
