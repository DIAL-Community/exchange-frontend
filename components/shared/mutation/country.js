import { gql } from '@apollo/client'

export const DELETE_COUNTRY = gql`
  mutation DeleteCountry($id: ID!) {
    deleteCountry(id: $id) {
      country {
        id
        slug
        name
      }
      errors
    }
  }
`

export const CREATE_COUNTRY = gql`
  mutation CreateCountry(
    $name: String!
    $slug: String!
    $description: String
  ) {
    createCountry(
      name: $name
      slug: $slug
      description: $description
    ) {
        country {
          id
          name
          slug
          code
          codeLonger
          description
        }
        errors
      }
    }
`
