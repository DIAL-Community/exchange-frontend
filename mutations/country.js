import { gql } from '@apollo/client'

export const DELETE_COUNTRY = gql`
  mutation DeleteCountry(
    $id: ID!
  ) {
    deleteCountry(
      id: $id
    ) {
      country {
       id
       slug
       name
      }
    }
  }
`

export const CREATE_COUNTRY = gql`
  mutation CreateCountry(
    $name: String!
    $slug: String!
    $code: String
    $codeLonger: String
  ) {
    createCountry(
      name: $name
      slug: $slug
      code: $code
      codeLonger: $codeLonger
    ) {
        country {
          slug
        }
        errors
      }
    }
`
