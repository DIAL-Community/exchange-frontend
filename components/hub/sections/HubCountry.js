import { useCallback } from 'react'
import { useIntl } from 'react-intl'
import { useQuery } from '@apollo/client'
import { Error, Loading, NotFound } from '../../shared/FetchStatus'
import { DPI_COUNTRY_DETAIL_QUERY } from '../../shared/query/country'
import HubCountryDetail from '../fragments/HubCountryDetail'
import HubBreadcrumb from './HubBreadcrumb'

const HubCountry = ({ slug }) => {
  const { formatMessage } = useIntl()
  const format = useCallback((id, values) => formatMessage({ id }, values), [formatMessage])

  const { loading, error, data } = useQuery(DPI_COUNTRY_DETAIL_QUERY, {
    variables: { slug }
  })

  if (loading) {
    return <Loading />
  } else if (error) {
    return <Error />
  } else if (!data?.country) {
    return <NotFound />
  }

  const { country } = data

  const slugNameMapping = (() => {
    const map = {}
    map[country.slug] = country.name

    return map
  })()

  return (
    <div className='flex flex-col'>
      <img className='h-80 w-full object-cover' alt='DIAL DPI Resource Hub' src='/images/hero-image/hub-hero.svg'/>
      <div className='absolute w-full left-1/2 -translate-x-1/2 min-h-[20rem]' style={{ top: 'var(--ui-header-height)' }}>
        <div className='max-w-catalog mx-auto'>
          <div className='px-4 lg:px-8 xl:px-56 text-dial-gray'>
            <HubBreadcrumb slugNameMapping={slugNameMapping} />
          </div>
        </div>
        <div className='text-2xl text-center text-dial-cotton pt-20 pb-4 mx-auto max-w-prose'>
          {country.name}
        </div>
        <img
          src={`https://flagcdn.com/${country.code.toLowerCase()}.svg`}
          alt={format('ui.country.logoAlt', { countryName: country.code })}
          className='h-16 mx-auto'
        />
      </div>
      <HubCountryDetail country={country} />
    </div>
  )
}

export default HubCountry
