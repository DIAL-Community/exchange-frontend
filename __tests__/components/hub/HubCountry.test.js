import { screen } from '@testing-library/dom'
import { FilterProvider } from '../../../components/context/FilterContext'
import { QueryParamContextProvider } from '../../../components/context/QueryParamContext'
import { ResourceFilterProvider } from '../../../components/context/ResourceFilterContext'
import HubCountry from '../../../components/hub/sections/HubCountry'
import { COUNTRIES_WITH_RESOURCES_SEARCH_QUERY, DPI_COUNTRY_DETAIL_QUERY } from '../../../components/shared/query/country'
import {
  PAGINATED_RESOURCES_QUERY, RESOURCE_PAGINATION_ATTRIBUTES_QUERY, RESOURCE_TYPE_SEARCH_QUERY
} from '../../../components/shared/query/resource'
import { render } from '../../test-utils'
import CustomMockedProvider, { generateMockApolloData } from '../../utils/CustomMockedProvider'
import { mockNextUseRouter, mockTenantApi } from '../../utils/nextMockImplementation'
import {
  countriesWithResources, countryResources, policyResourcePagination, policyResources, websiteResourcePagination,
  websiteResources
} from './data/HubCountry.data'
import { resourceTypes } from './data/HubResourceFinder.data'

mockTenantApi()
mockNextUseRouter()
describe('Unit tests for the opportunity detail page.', () => {
  const mockPolicyResources = generateMockApolloData(
    PAGINATED_RESOURCES_QUERY,
    {
      'search': '',
      'countries': ['15'],
      'resourceTypes': ['Government Document'],
      'limit': 6,
      'offset': 0
    },
    null,
    policyResources
  )

  const mockPolicyResourcePagination = generateMockApolloData(
    RESOURCE_PAGINATION_ATTRIBUTES_QUERY,
    {
      'search': '',
      'countries': ['15'],
      'resourceTypes': ['Government Document']
    },
    null,
    policyResourcePagination
  )

  const mockWebsiteResources = generateMockApolloData(
    PAGINATED_RESOURCES_QUERY,
    {
      'search': '',
      'countries': ['15'],
      'resourceTypes': ['National Website'],
      'limit': 6,
      'offset': 0
    },
    null,
    websiteResources
  )

  const mockWebsiteResourcePagination = generateMockApolloData(
    RESOURCE_PAGINATION_ATTRIBUTES_QUERY,
    {
      'search': '',
      'countries': ['15'],
      'resourceTypes': ['National Website']
    },
    null,
    websiteResourcePagination
  )

  const mockResources = generateMockApolloData(
    PAGINATED_RESOURCES_QUERY,
    {
      'search': '',
      'countries': ['15'],
      'resourceTypes': [
        'Article',
        'Blog',
        'Book',
        'Case Study',
        'Essay',
        'Expert Comment',
        'Explainer',
        'Guidance',
        'Playbook',
        'Report',
        'Spotlight',
        'Unspecified Type',
        'White Paper',
        'Working Papers'
      ],
      'limit': 6,
      'offset': 0
    },
    null,
    websiteResources
  )

  const mockResourcePagination = generateMockApolloData(
    RESOURCE_PAGINATION_ATTRIBUTES_QUERY,
    {
      'search': '',
      'countries': ['15'],
      'resourceTypes': [
        'Article',
        'Blog',
        'Book',
        'Case Study',
        'Essay',
        'Expert Comment',
        'Explainer',
        'Guidance',
        'Playbook',
        'Report',
        'Spotlight',
        'Unspecified Type',
        'White Paper',
        'Working Papers'
      ]
    },
    null,
    websiteResourcePagination
  )

  const mockCountry = generateMockApolloData(
    DPI_COUNTRY_DETAIL_QUERY,
    {
      'slug': 'zambia'
    },
    null,
    countryResources
  )

  const mockCountriesWithResources = generateMockApolloData(
    COUNTRIES_WITH_RESOURCES_SEARCH_QUERY,
    {
      'search': ''
    },
    null,
    countriesWithResources
  )

  const mockResourceTypes = generateMockApolloData(
    RESOURCE_TYPE_SEARCH_QUERY,
    {
      'search': ''
    },
    null,
    resourceTypes
  )

  test('Should render detail of a country detail page.', async () => {
    const { container } = render(
      <CustomMockedProvider
        mocks={[
          mockCountry,
          mockCountriesWithResources,
          mockPolicyResources,
          mockPolicyResourcePagination,
          mockWebsiteResources,
          mockWebsiteResourcePagination,
          mockResources,
          mockResourcePagination,
          mockResourceTypes
        ]}
      >
        <QueryParamContextProvider>
          <ResourceFilterProvider>
            <FilterProvider>
              <HubCountry slug='zambia' />
            </FilterProvider>
          </ResourceFilterProvider>
        </QueryParamContextProvider>
      </CustomMockedProvider>
    )

    expect(await screen.findByText('Zambia')).toBeInTheDocument()
    expect(container).toMatchSnapshot()

    expect(await screen.findByText('Key Policies, Legal and Regulatory Frameworks')).toBeInTheDocument()
    expect(await screen.findByText('Sierra Leone National Cyber Security Strategy (2021-2025)')).toBeInTheDocument()
    expect(await screen.findByText('Key Government Websites')).toBeInTheDocument()
  })
})
