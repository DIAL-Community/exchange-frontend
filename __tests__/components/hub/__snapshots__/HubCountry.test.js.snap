// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unit tests for the opportunity detail page. Should render detail of a country detail page. 1`] = `
<div>
  <div
    class="Toastify"
  />
  <div
    class="flex flex-col"
  >
    <img
      alt="DIAL DPI Resource Hub"
      class="h-32 w-full object-cover"
      src="/images/hero-image/hub-hero.svg"
    />
    <div
      class="absolute w-full left-1/2 -translate-x-1/2"
    >
      <div
        class="max-w-catalog mx-auto py-2"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 text-dial-gray"
        >
          <div
            class="whitespace-nowrap text-ellipsis overflow-hidden text-sm"
          >
            <a
              class="inline"
              href="/"
            >
              Home
            </a>
          </div>
        </div>
      </div>
      <div
        class="max-w-catalog mx-auto py-1"
      >
        <div
          class="flex gap-4 px-4 lg:px-8 xl:px-56"
        >
          <img
            alt="Flag for country: ZM."
            class="h-16"
            src="https://flagcdn.com/zm.svg"
          />
          <div
            class="text-2xl text-dial-cotton my-auto"
          >
            Zambia
          </div>
        </div>
      </div>
    </div>
    <div
      class="flex flex-col"
    >
      <div
        class="description-section pb-4"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 text-dial-stratos flex flex-col"
        >
          <div
            class="flex flex-flex-wrap gap-3"
          >
            <div
              class="text-xl font-medium py-6 "
            >
              Country Overview
            </div>
          </div>
          <div
            class="relative"
          >
            <div
              class="html-react-parser-viewer text-base"
            >
              Some description
            </div>
          </div>
        </div>
      </div>
      <div
        class="policy-section bg-dial-sapphire"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 text-dial-cotton flex flex-col"
        >
          <div
            class="text-xl font-medium py-6"
          >
            Key Policies, Legal and Regulatory Frameworks
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <div
              class="flex flex-col gap-y-4"
            >
              <hr
                class="border-b border-gray-300 border-dashed"
              />
              <div
                class="relative"
              >
                <a
                  href="/hub/countries/zm/resources/sierra-leone-national-data-strategy-2023"
                >
                  <div
                    class="rounded-lg min-h-[10rem] group"
                  >
                    <div
                      class="flex flex-col gap-y-3"
                    >
                      <div
                        class="text-lg font-medium flex"
                      >
                        <span
                          class="border-b border-transparent group-hover:border-dial-cotton"
                        >
                          Sierra Leone National Data Strategy (2023)
                        </span>
                      </div>
                      <div
                        class="line-clamp-4 text-justify"
                      >
                        Some parsed description
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="flex flex-col gap-y-4"
            >
              <hr
                class="border-b border-gray-300 border-dashed"
              />
              <div
                class="relative"
              >
                <a
                  href="/hub/countries/zm/resources/sierra-leone-national-cyber-security-strategy-2021-2025"
                >
                  <div
                    class="rounded-lg min-h-[10rem] group"
                  >
                    <div
                      class="flex flex-col gap-y-3"
                    >
                      <div
                        class="text-lg font-medium flex"
                      >
                        <span
                          class="border-b border-transparent group-hover:border-dial-cotton"
                        >
                          Sierra Leone National Cyber Security Strategy (2021-2025)
                        </span>
                      </div>
                      <div
                        class="line-clamp-4 text-justify"
                      >
                        Some parsed description
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="flex flex-col gap-y-4"
            >
              <hr
                class="border-b border-gray-300 border-dashed"
              />
              <div
                class="relative"
              >
                <a
                  href="/hub/countries/zm/resources/sierra-leone-national-cyber-security-policy-2021"
                >
                  <div
                    class="rounded-lg min-h-[10rem] group"
                  >
                    <div
                      class="flex flex-col gap-y-3"
                    >
                      <div
                        class="text-lg font-medium flex"
                      >
                        <span
                          class="border-b border-transparent group-hover:border-dial-cotton"
                        >
                          Sierra Leone National Cyber Security Policy (2021)
                        </span>
                      </div>
                      <div
                        class="line-clamp-4 text-justify"
                      >
                        Some parsed description
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
            <div
              class="flex flex-col gap-y-4"
            >
              <hr
                class="border-b border-gray-300 border-dashed"
              />
              <div
                class="relative"
              >
                <a
                  href="/hub/countries/zm/resources/sierra-leone-national-innovation--digital-strategy"
                >
                  <div
                    class="rounded-lg min-h-[10rem] group"
                  >
                    <div
                      class="flex flex-col gap-y-3"
                    >
                      <div
                        class="text-lg font-medium flex"
                      >
                        <span
                          class="border-b border-transparent group-hover:border-dial-cotton"
                        >
                          Sierra Leone National Innovation & Digital Strategy
                        </span>
                      </div>
                      <div
                        class="line-clamp-4 text-justify"
                      >
                        Some parsed description
                      </div>
                    </div>
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6 text-dial-slate-300"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col md:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm"
              >
                Showing 1 - 4 of 4 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm"
                role="navigation"
              >
                <li
                  class="previous opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="text-dial-cotton font-semibold"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="next opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="next"
                    role="button"
                    tabindex="-1"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
      <div
        class="website-section"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 text-dial-stratos flex flex-col"
        >
          <div
            class="text-xl font-medium py-6"
          >
            Key Government Websites
          </div>
          <div
            class="grid md:grid-cols-2 gap-4"
          >
            <div
              class="border bg-dial-yellow rounded-md min-h-[2rem]"
            >
              <div
                class="flex flex-col"
              >
                <a
                  class="p-3 text-lg font-medium flex group"
                  href="/hub/countries/zm/resources/stats-sl-open-data-dashboard"
                >
                  <div
                    class="line-clamp-1 border-b border-transparent group-hover:border-dial-stratos"
                  >
                    Stats SL Open Data Dashboard
                  </div>
                </a>
                <a
                  class="px-3 pb-3 text-justify text-sm flex group"
                  href="//www.statistics.sl/index.php/what-we-offer/open-data-free-datasets.html"
                  rel="noreferrer"
                  target="_blank"
                >
                  <div
                    class="line-clamp-1 border-b border-transparent group-hover:border-dial-stratos"
                  >
                    www.statistics.sl/index.php/what-we-offer/open-data-free-datasets.html
                     ⧉
                  </div>
                </a>
              </div>
            </div>
            <div
              class="border bg-dial-yellow rounded-md min-h-[2rem]"
            >
              <div
                class="flex flex-col"
              >
                <a
                  class="p-3 text-lg font-medium flex group"
                  href="/hub/countries/zm/resources/sierra-leone-data-portal-website"
                >
                  <div
                    class="line-clamp-1 border-b border-transparent group-hover:border-dial-stratos"
                  >
                    Sierra Leone Data Portal Website
                  </div>
                </a>
                <a
                  class="px-3 pb-3 text-justify text-sm flex group"
                  href="//sierraleone.opendataforafrica.org/data/#menu=topic"
                  rel="noreferrer"
                  target="_blank"
                >
                  <div
                    class="line-clamp-1 border-b border-transparent group-hover:border-dial-stratos"
                  >
                    sierraleone.opendataforafrica.org/data/#menu=topic
                     ⧉
                  </div>
                </a>
              </div>
            </div>
            <div
              class="border bg-dial-yellow rounded-md min-h-[2rem]"
            >
              <div
                class="flex flex-col"
              >
                <a
                  class="p-3 text-lg font-medium flex group"
                  href="/hub/countries/zm/resources/sierra-leone-citizen-portal"
                >
                  <div
                    class="line-clamp-1 border-b border-transparent group-hover:border-dial-stratos"
                  >
                    Sierra Leone Citizen Portal
                  </div>
                </a>
                <a
                  class="px-3 pb-3 text-justify text-sm flex group"
                  href="//portal.moppa.gov.sl"
                  rel="noreferrer"
                  target="_blank"
                >
                  <div
                    class="line-clamp-1 border-b border-transparent group-hover:border-dial-stratos"
                  >
                    portal.moppa.gov.sl
                     ⧉
                  </div>
                </a>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6 false"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col md:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm"
              >
                Showing 1 - 3 of 3 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm"
                role="navigation"
              >
                <li
                  class="previous opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="false font-semibold"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="next opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="next"
                    role="button"
                    tabindex="-1"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
      <div
        class="product-section bg-dial-sapphire"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 flex flex-col"
        >
          <div
            class="text-xl font-medium text-center py-6 text-dial-cotton"
          >
            Products
          </div>
          <div
            class="grid grid-cols-1 md:grid-cols-2 xl:grid-cols-3 gap-8"
          >
            <div
              class="relative"
            >
              <a
                href="/products/rapidpro"
              >
                <div
                  class="flex flex-col gap-2"
                >
                  <div
                    class="min-h-[12rem] flex items-center justify-center bg-white"
                  >
                    <img
                      alt="Logo of Product."
                      class="object-contain max-h-[6rem]"
                      src="http://localhost:3000//assets/products/rapidpro.png"
                    />
                  </div>
                  <div
                    class="text-center text-white"
                  >
                    RapidPro
                  </div>
                </div>
              </a>
              <div
                class="absolute top-2 right-2"
              />
            </div>
            <div
              class="relative"
            >
              <a
                href="/products/kobotoolbox"
              >
                <div
                  class="flex flex-col gap-2"
                >
                  <div
                    class="min-h-[12rem] flex items-center justify-center bg-white"
                  >
                    <img
                      alt="Logo of Product."
                      class="object-contain max-h-[6rem]"
                      src="http://localhost:3000//assets/products/kobotoolbox.png"
                    />
                  </div>
                  <div
                    class="text-center text-white"
                  >
                    KoboToolbox
                  </div>
                </div>
              </a>
              <div
                class="absolute top-2 right-2"
              />
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6 text-dial-slate-300"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col md:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm"
              >
                Showing 1 - 2 of 2 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm"
                role="navigation"
              >
                <li
                  class="previous opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="text-dial-cotton font-semibold"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="next opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="next"
                    role="button"
                    tabindex="-1"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
      <div
        class="resource-section"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 flex flex-col"
        >
          <div
            class="text-xl font-medium py-6"
          >
            Resources, Reports, and Articles
          </div>
          Fetching data ...
          Fetching data ...
        </div>
      </div>
      <div
        class="candidate-resource-section bg-dial-lavender"
      >
        <div
          class="px-4 lg:px-8 xl:px-56 flex flex-col text-dial-stratos"
        >
          <div
            class="flex flex-col md:flex-row gap-3"
          >
            <div
              class="text-xl font-medium py-6"
            >
              Suggested Resources, Reports, and Articles
            </div>
            <div
              class="ml-auto text-sm flex justify-center items-center"
            >
              <a
                class="cursor-pointer bg-dial-sapphire px-4 py-2 rounded "
                href="/hub/resources/suggest"
              >
                <div
                  class="flex flex-row gap-1 text-dial-cotton"
                >
                  <svg
                    class="inline my-auto"
                    fill="none"
                    height="1em"
                    stroke="currentColor"
                    stroke-linecap="round"
                    stroke-linejoin="round"
                    stroke-width="2"
                    viewBox="0 0 24 24"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <circle
                      cx="12"
                      cy="12"
                      r="10"
                    />
                    <line
                      x1="12"
                      x2="12"
                      y1="8"
                      y2="16"
                    />
                    <line
                      x1="8"
                      x2="16"
                      y1="12"
                      y2="12"
                    />
                  </svg>
                  Create New
                </div>
              </a>
            </div>
          </div>
          <div
            class="flex flex-col gap-2"
          >
            <div
              class="flex flex-col gap-y-4"
            >
              <hr
                class="border-b border-gray-300 border-dashed"
              />
              <div
                class="relative"
              >
                <div
                  class="rounded-lg min-h-[6rem] group"
                >
                  <div
                    class="flex flex-col gap-y-3 relative"
                  >
                    <div
                      class="text-lg font-medium flex"
                    >
                      <span
                        class="border-b border-transparent group-hover:border-dial-stratos"
                      >
                        Test Adding More Candidate Resource
                      </span>
                    </div>
                    <div
                      class="absolute top-2 right-2 bg-dial-sapphire rounded"
                    >
                      <div
                        class="text-dial-cotton text-xs px-2 py-1"
                      >
                        In Review
                      </div>
                    </div>
                    <div
                      class="line-clamp-2 text-justify text-sm"
                    >
                      Parsed example of description
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
          <div
            class="flex flex-col gap-y-4 my-6 false"
          >
            <hr
              class="border-b border-dial-slate-300"
            />
            <div
              class="flex flex-col md:flex-row gap-x-16 gap-y-3"
            >
              <div
                class="my-auto text-sm"
              >
                Showing 1 - 1 of 1 total
              </div>
              <ul
                aria-label="Pagination"
                class="flex gap-x-3 text-sm"
                role="navigation"
              >
                <li
                  class="previous opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Previous page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="prev"
                    role="button"
                    tabindex="-1"
                  >
                    &lt;
                  </a>
                </li>
                <li
                  class="false font-semibold"
                >
                  <a
                    aria-current="page"
                    aria-label="Page 1 is your current page"
                    class="block w-10 h-10 leading-10 text-center"
                    rel="canonical"
                    role="button"
                    tabindex="-1"
                  >
                    1
                  </a>
                </li>
                <li
                  class="next opacity-30"
                >
                  <a
                    aria-disabled="true"
                    aria-label="Next page"
                    class="block w-10 h-10 leading-10 text-center "
                    rel="next"
                    role="button"
                    tabindex="-1"
                  >
                    &gt;
                  </a>
                </li>
              </ul>
            </div>
            <hr
              class="border-b border-dial-slate-300"
            />
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
